### PR TITLE
add source linking

### PR DESCRIPTION
**Pull request by paulomorgado**
### The issue or feature being addressed

Adding source linking to allow step into sources when debugging.

### Confirm the following

- [X]  I have merged the latest changes from the dev vX.Y branch
- [X]  I have successfully run a local build
- [X]  I have included unit tests for the issue/feature
- [ ]  I have targeted the PR against the latest dev vX.Y branch

### CODE REVIEW

**Comment by paulomorgado** [comment]
The strong name signer is removing the source linking information from the PDBs.

**Comment by paulomorgado** [comment]
https://github.com/brutaldev/StrongNameSigner/issues/50

**Comment by paulomorgado** [comment]
@reisenberger, compare the size of the PDBs under the build folder with the size of the PDBs under the bin folder. Is it the same?

Also, the PDB must contain the string https://github.com/App-vNext/Polly/ somewhere in it.

I've been looking at how [System.Reactive](https://github.com/dotnet/reactive) is built to see if I can set the same thing up here.

**Comment by paulomorgado** [comment]
Maybe we should fix #504 first. Do you have any clue to why the tests are failing for .NET Framework runtimes?

**Comment by reisenberger** [comment]
> @reisenberger, compare 

@paulomorgado. Yes, my yesterday's reply was based on a quick read of the correspondence across the tool projects.  I see, on running it up, what you are saying.

So the [cecil proj is saying this is fixed](https://github.com/jbevain/cecil/pull/432), Brutal.Dev.StrongNameSigner is [referencing the latest version of Cecil](https://github.com/brutaldev/StrongNameSigner/blob/f4063dd86ce7d5a17a44e719cb23871d45877767/src/Brutal.Dev.StrongNameSigner/packages.config#L3) but we are still experiencing the problem.

We are not tied to using [brutaldev/StrongNameSigner](https://github.com/brutaldev/StrongNameSigner) for strong-naming; that was useful for packaging when we were shipping both non-strong-named and strong-named, but we now ship strong-named only.

Thank you for everything you are doing on this!

**Comment by reisenberger** [comment]
> #504 Do you have any clue to why the tests are failing for .NET Framework runtimes?

Out of time to look at that today, but we can dig deeper soon.

**Comment by reisenberger** [comment]
@paulomorgado We can remove Brutal.Dev.StrongNameSigner from the mix with a [commit like this](https://github.com/reisenberger/Polly/commit/8010d054c6364f6591c10aac1dfb69799809155a).

**Comment by reisenberger** [comment]
I worked through a bunch of further issues with adding SourceLink.  It suggests:

+ we need to remove Brutal.Dev.StrongNameSigner per my [previous comment](https://github.com/App-vNext/Polly/pull/502#issuecomment-423057926)
+ we need to replace [direct call to nuget pack](https://github.com/App-vNext/Polly/blob/master/build.cake#L261-L268) with [dotnet pack](https://cakebuild.net/api/Cake.Common.Tools.DotNetCore.Pack/), as inserting the new `<repository commit="{specific-sha}">` attribute in the nupkg relies on this, afaics
+ for `dotnet pack` to play well with the multi-targeting, we likely want to merge #504 first; and will have to change a bunch of other parts of the build script

In general, given Polly no longer targets any .NET Framework targets since v6.0, it would be an opportunity to overhaul the cake build script and move it to dotnet cli throughout.

*TL;DR* This PR sits [behind #504](https://github.com/App-vNext/Polly/pull/504#issuecomment-437652822).

**Comment by paulomorgado** [comment]
Thanks for the update, @reisenberger. Let me know if you need anything from me.

**Comment by reisenberger** [comment]
This PR was unexpectedly automatically closed by github, when deleting a (now-outdated) base branch which the PR targeted .  

We still intend to add source linking, though it will now have to be based off the new Polly v7.0.0, and address the issues [previously identified here](https://github.com/App-vNext/Polly/pull/502#issuecomment-437653893)

Thanks @paulomorgado for all you originally did on this! We will revert when we are ready to add source linking. I will be reviewing next work items after Polly v7.0.0 and [Simmy](https://github.com/App-vNext/Simmy) are both released.
